### PR TITLE
Microsoft CAS/first commit before unit-test and test with UI

### PR DESCRIPTION
**Pull request by obenacot**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/33998

## Description
Added the required functionality. 
Not Finshied - 
1. Before major testing in microsoft API
2. code review by Shai.
3. Unit testing.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftCloudAppSecurity/MicrosoftCloudAppSecurity.yml [hunk 1]
```diff
@@ -294,10 +311,235 @@ script:
...
+      secret: false
+    - default: false
+      description: The email address of the user
+      isArray: false
+      name: contactEmail
+      required: false
+      secret: false
+    deprecated: false
+    description: An alert on a suspicious but not malicious activity, such as a penetration
+      test or other authorized suspicious action
```
Comment (yaakovi) [inline Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftCloudAppSecurity/MicrosoftCloudAppSecurity.yml:392]: improve the description

File: Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftCloudAppSecurity/MicrosoftCloudAppSecurity.yml [hunk 2]
```diff
@@ -294,10 +311,235 @@ script:
...
+      description: |-
+        Multiple alerts matching the specified filters.
+        Alert_id should appear similar to - "1234567890abcdefg".
+        Mandatory, unless you use a custom filter.
+      isArray: true
+      name: alert_ids
+      required: false
+      secret: false
+    - default: false
+      description: A custom filter by which to filter the returned files. If you pass
```
Comment (yaakovi) [inline Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftCloudAppSecurity/MicrosoftCloudAppSecurity.yml:406]: * note that we expect a JSON value here
* add an example

File: Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftCloudAppSecurity/MicrosoftCloudAppSecurity.yml [hunk 3]
```diff
@@ -294,10 +311,235 @@ script:
...
+
+        0: Not of interest
+        1: Too many similar alerts
+        3: Alert is not accurate
+        4: Other
+      isArray: false
+      name: reasonId
+      predefined:
+      - Not of interest
+      - ' Too many similar alerts'
```
Comment (yaakovi) [inline Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftCloudAppSecurity/MicrosoftCloudAppSecurity.yml:498]:
```suggestion
      - Too many similar alerts
```

**Review by yaakovi — CHANGES_REQUESTED** [review]
update the integration README with the new commands and the deprecation of the old commands

File: Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftCloudAppSecurity/MicrosoftCloudAppSecurity.py [hunk 4]
```diff
@@ -217,7 +257,36 @@ def build_filter_and_url_to_search_with(url_suffix: str, custom_filter: Optional
...
+    request_data['feedbackText'] = feedbackText
+    request_data['allowContact'] = allowContact
+    request_data['contactEmail'] = contactEmail
+
+    return request_data
+
+
 def args_to_filter_for_dismiss_and_resolve_alerts(alert_ids: Any, custom_filter: Any, comments: Any):
+    """
+    Deprecated by args_to_filter_close_alerts.
```
[the remark this answers is not in the document]
Comment (obenacot) [reply Packs/MicrosoftCloudAppSecurity/Integrations/MicrosoftCloudAppSecurity/MicrosoftCloudAppSecurity.py:288]: Function has 4 usages, all by deprecated functions.

**Comment by obenacot** [comment]
Somehow 697 commits from master (instead of 17) has been inserted to the PR. 
Creating a new PR instead:

**Pull request closed without merging**